packages/x-date-pickers/src/YearCalendar/YearCalendar.tsx

```diff
@@ -221,22 +222,25 @@ export const YearCalendar = React.forwardRef(function YearCalendar<TDate extends
     );
   }, [selectedYear]);
 
+  const horizontalDirection = theme.direction === 'ltr' && !yearsReversed ? 1 : -1;
+  const verticalDirection = !yearsReversed ? yearsPerRow * 1 : yearsPerRow * -1;
+
   const handleKeyDown = useEventCallback((event: React.KeyboardEvent, year: number) => {
     switch (event.key) {
       case 'ArrowUp':
-        focusYear(year - yearsPerRow);
+        focusYear(year - verticalDirection);
         event.preventDefault();
         break;
       case 'ArrowDown':
-        focusYear(year + yearsPerRow);
+        focusYear(year + verticalDirection);
         event.preventDefault();
         break;
       case 'ArrowLeft':
-        focusYear(year + (theme.direction === 'ltr' ? -1 : 1));
+        focusYear(year - horizontalDirection);
```

Comment:
Hey @LukasTy, sorry about the late response.
The existing one seems more natural to me, it feels a bit odd to press the right arrow but focus moving to the right.

Reply:
@MBilalShafi Thank you for the confirmation. 🙏

Reply:
Hi all, sorry for the delay but it's great to see that this has moved forward and I also had questions about the RTL usability. One thing that seem like a good idea is to separate the direction into a param, but wasn't sure if it should be flipped or not, great to get feedback!

Reply:
I think I sorted out the horizontalDirection now so that it works properly in the different cases. I tested isRtl by wrapping the demo in a div with dir="rtl" and the icon and text field value were reversed. The keyboard routines seemed to work correctly. Perhaps @MBilalShafi could test the latest version still?

Reply:
Looks OK now 👍